go/arrow/csv/reader.go

```diff
@@ -740,81 +740,67 @@ func (r *Reader) parseDate32(field array.Builder, str string) {
 	field.(*array.Date32Builder).Append(arrow.Date32FromTime(tm))
 }
 
-func (r *Reader) parseTime32(field array.Builder, str string, unit arrow.TimeUnit) {
+func (r *Reader) parseDate64(field array.Builder, str string) {
 	if r.isNull(str) {
 		field.AppendNull()
 		return
 	}
 
-	val, err := arrow.Time32FromString(str, unit)
+	tm, err := time.Parse("2006-01-02", str)
```

Comment:
do we want to allow specifying the date format?

Reply:
I don't think so, as the marshalling to csv uses this format (same as date32, so out of scope?)

Reply:
sure. if anyone ends up wanting that it can just be a follow-up enhancement